core/src/main/scala/org/apache/spark/BarrierCoordinator.scala

```diff
@@ -199,14 +268,20 @@ private[spark] class BarrierCoordinator(
   }
 
   override def receiveAndReply(context: RpcCallContext): PartialFunction[Any, Unit] = {
-    case request @ RequestToSync(numTasks, stageId, stageAttemptId, _, _) =>
+    case request @ BarrierRequestToSync(numTasks, stageId, stageAttemptId, _, _, _) =>
       // Get or init the ContextBarrierState correspond to the stage attempt.
       val barrierId = ContextBarrierId(stageId, stageAttemptId)
       states.computeIfAbsent(barrierId,
         (key: ContextBarrierId) => new ContextBarrierState(key, numTasks))
       val barrierState = states.get(barrierId)
-
-      barrierState.handleRequest(context, request)
+      barrierState.handleBarrierRequest(context, request)
+    case request @ AllGatherRequestToSync(numTasks, stageId, stageAttemptId, _, _, _, _) =>
```

Comment:
The case is not necessary here, we only need to switch the case when we are to fetch allGather messages.

Reply:
The main problem is Scala does not allow variables in alternative patterns. Any thoughts on a workaround?

Reply:
we can keep the `barrierState.handleRequest(context, request)` here. And then inside the `handleRequest` method, we can have:
```
val allGatherMessage = request match {
    case ag: AllGatherRequestToSync => ag. allGatherMessage
    case _ => Array[Byte]()
}
```

Reply:
resolved!